### DISCUSSION TITLE
copilot completion requires long wait

### DISCUSSION THREAD

**Discussion by xuhdev** [emacs-lsp/lsp-mode#4655] [Q&A] [answered]
I've enabled the copilot completion. But it looks like I have to remain idle for 3 to 4 seconds for completion to show up. Is there a parameter that I should tune to make this wait time shorter? My `lsp-inline-completion-idle-delay` and `lsp-idle-delay` are 0.5 but they don't seem to do anything.

**Comment by xuhdev** [comment, marked as answer]
I found that in some circumstances the completion jumps out quite fast. Perhaps it's just how copilot works and it sometimes is fast and sometimes slow.

**Comment by kassick** [comment]
To complement this:
Inline completions *on idle* happen after `(+ lsp-idle-delay lsp-inline-completion-idle-delay)` . That's 2.5s by default.
@xuhdev's choice of reducing the value of `lsp-inline-completion-idle-delay` is the way to go, but one must take into account `lsp-idle-delay` as well.

That said, the copilot server may take a while to respond, specially on the first requests. If you trigger inline completions manually (you can bind `lsp-inline-completion-display` to some key to force triggering), you'll notice that the server sometimes takes a few seconds to come up with a suggestion, sometimes its quite snappy.
